platform/mellanox/cmis_host_mgmt/cmis_host_mgmt.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2021-2024 NVIDIA CORPORATION & AFFILIATES.
+# Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import shutil
+import click
+import re
+import os
+import subprocess
+
+
+class CMISHostMgmtActivator:
+    PARAMS = {
+        "sai_profile": {
+            "file_name": "sai.profile",
+            "enabled_param": "SAI_INDEPENDENT_MODULE_MODE=1",
+            "disabled_param": "SAI_INDEPENDENT_MODULE_MODE=0"
+        },
+        "pmon_daemon_control": {
+            "file_name": "pmon_daemon_control.json",
+            "enabled_param": "\"skip_xcvrd_cmis_mgr\": false",
+            "disabled_param": "\"skip_xcvrd_cmis_mgr\": true",
+        },
+        "sai_xml": {
+            "file_name": "sai_<>.xml", # will be filled at main, since we can't know the SKU here 
+            "enabled_param": "<late-create-all-ports>1</late-create-all-ports>",
+            "disabled_param": "<late-create-all-ports>1</late-create-all-ports>" # Shouldn't be called
+        }
+    }
+
+    @staticmethod
+    def change_param(param, path, action):
+        file_path = '{}/{}'.format(path, CMISHostMgmtActivator.PARAMS[param]["file_name"])
+        lines = None
+
+        with open(file_path, 'r') as param_file:
```

Comment:
why not to open the file from the beginning as 'rw'?

Reply:
Fixed